6238_prototype/src/main/java/frc/robot/commands/ShooterCommand.java

```diff
@@ -5,22 +5,67 @@
 import frc.robot.subsystems.BallSubsystem;
 
 public class ShooterCommand extends CommandBase{
+    public static enum PneumaticKickers {LEFT_KICKER, RIGHT_KICKER, BOTH};
+
+    private final PneumaticKickers kicker;
     private final BallSubsystem ball; 
-    private final SmartDashboardParam upperShooterSpeedTarget;
-    public ShooterCommand(BallSubsystem ball) {
+    private final double upperShooterRPMTarget;
+    private long timerStart;
+    private final long timeLimit;
+    private final double rpmDiffTolerance;
+    private boolean pneumaticsActivated;
+    private long countdownStart;
+
+    public ShooterCommand(BallSubsystem ball, PneumaticKickers kicker, double rpmTarget) {
         this.ball = ball;
-        upperShooterSpeedTarget = new SmartDashboardParam("upperMotorSpeedTarget");
+        upperShooterRPMTarget = rpmTarget;
```

Comment:
This should still be attached to smart dashboard so we can tune the distance.  We probably want a hard code distance when using this command in autonomous.  I think the way to handle this is probably to have a subclass that update the RPM target based on a SmartDashboardParam.  You'll need to make sure you read the param on each call to `execute`